FlightAxis/flightaxis.py

```diff
@@ -201,6 +222,66 @@ def disableRC(self, doPrint=False) -> bool:
             print(response.content)
         return response.ok
 
+    def saveStates(self):
+        with open(self.filename, mode='a') as f:
+            writer = csv.writer(f, delimiter=',')
+            writer.writerows(self.rows)
+
+
+
+    def updateActuators(self,values):
+        """
+        Incoming:
+        rc_in idx -> SOAP order
+        0 -> yaw (3)
+        1 -> throttle (2)
+        2 -> roll (0) # invert
+        3 -> pitch (1)
+
+        Desired:
+        0 -> roll
+        1 -> pitch
+        2 -> yaw
+        3 -> throttle
+        
+        """
+         # Condition channel
+        #     # -0.66..0.66 to 0-1
+        RMIN=-0.66
+        RMAX=0.66
+        TMIN=0.0
+        TMAX=1.0
+        YAW_IDX = (0,3,-1)
+        THROTTLE_IDX = (1,2,1)
+        ROLL_IDX = (2,0,-1)
+        PITCH_IDX = (3,1,1)
+        GEAR_IDX = (4,6,1)
+        MODE_IDX = (5,7,1)
+        # yaw
+        val = (YAW_IDX[2]*values[YAW_IDX[0]]-RMIN)/(RMAX-RMIN)*(TMAX-TMIN)+TMIN
```

Comment:
I would move this calculation as a method. Also, it might be convenient to structure `bounds = (min, max)` so that it can be passed more easily.

Reply:
Fixed